### PR TITLE
Add typings for project

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: minor
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		provisioningApiKey: string;
+		apiEndpoint: string;
+	}
+
+	interface ResinRegisterDevice {
+		generateUniqueKey: () => string;
+		register: (
+			regOpts: RegisterOpts,
+			cb: (err: Error | undefined, deviceInfo: any) => void,
+		) => Promise<string>;
```
Comment (zvin) [inline index.d.ts]: This is a Bluebird, not a Promise

Comment (CameronDiver) [reply index.d.ts]: Good catch!

File: package.json [hunk 2]
```diff
@@ -1,53 +1,54 @@
...
-    "sinon": "^1.17.6"
-  },
-  "dependencies": {
-    "bluebird": "^3.0.0",
-    "randomstring": "^1.1.5"
-  },
-  "peerDependencies": {
-    "resin-request": "^9.0.2"
-  }
+	"name": "resin-register-device",
```
Comment (Page-) [inline package.json]: This file should be space-indented as it's controlled by the `npm` cli; running something like `npm i`/`npm update` should reformat it automatically

Comment (CameronDiver) [reply package.json]: Oh man didn't even notice the entire file changing.

**Pull request closed without merging**